docs/intro/tutorial.rst

```diff
@@ -1,141 +1,434 @@
-list page on `books.toscrape.com <http://books.toscrape.com/>`_.
+.. _Python: https://docs.python.org/
 
-.. code-block:: python
 
-    from web_poet.pages import WebPage
+Create a project folder
+=======================
+
+web-poet does not limit how you structure your web-poet web data extraction
+code, beyond the limitations of Python itself.
+
+However, in this tutorial you will use a specific project folder structure
+designed with web-poet best practices in mind. Consider using the same project
+folder structure in all your web-poet projects.
+
+First create your project folder: ``tutorial-project``.
+
+Within ``tutorial-project``, create:
+
+-   A ``run.py`` file, a file specific to this tutorial where you will put code
+    to test the execution of your web data extraction code.
+
+-   A ``tutorial`` folder, where you will place your web data extraction code.
+
+Within ``tutorial-project/tutorial``, create:
+
+-   An ``__init__.py`` file, so that the ``tutorial`` folder becomes an
+    importable Python module.
+
+-   An ``items.py`` file, where you will define item classes to store extracted
+    data.
+
+-   A ``pages`` folder, where you will define your page object classes.
+
+Within ``tutorial-project/tutorial/pages``, create:
 
+-   An ``__init__.py`` file.
 
-    class BookLinksPage(WebPage):
+-   A ``books_toscrape_com.py`` file, for page object class code targeting
+    `books.toscrape.com`_.
 
-        @property
-        def links(self):
-            return self.css(".image_container a::attr(href)").getall()
+Your project folder should look as follows:
 
-        def to_item(self) -> dict:
-            return {
-                "links": self.links,
-            }
+.. code-block:: text
 
-Downloading Response
+    tutorial-project/
+        run.py
+        tutorial/
+            __init__.py
+            items.py
+            pages/
+                __init__.py
+                books_toscrape_com.py
+
+
+Create an item class
 ====================
 
-The ``BookLinksPage`` Page Object requires a
-:class:`~.HttpResponse` with the
-book list page content in order to extract the information we need. First,
-let's download the page using ``requests`` library.
+While it is possible to store the extracted data in a Python dictionary, it is
+a good practice to create an item class that:
+
+-   Defines the specific attributes that you aim to extract, triggering an
+    exception if you extract unintended attributes or fail to extract expected
+    attributes.
+
+-   Allows defining default values for some attributes.
+
+web-poet uses itemadapter_ for item class support, which means that any kind of
+item class can be used. In this tutorial, you will use attrs_ to define your
+item class.
+
+.. _attrs: https://www.attrs.org/en/stable/
+.. _itemadapter: https://github.com/scrapy/itemadapter
+
+Copy the following code into ``tutorial-project/tutorial/items.py``:
+
+.. literalinclude:: /../tutorial-project/tutorial/items.py
```

Comment:
What do you think about moving tutorial-project to the docs folder? The code looks a bit strange as a general-purpose example (imports in the middle of files, etc).